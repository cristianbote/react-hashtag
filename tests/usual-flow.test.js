import React from 'react';
import {configure, shallow, mount} from 'enzyme';
import ReactHashtag from '../src/index';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('ReactHashtag should behave', () => {

    it('Should render hashtags', () => {
        const context = shallow(
            <div>
                <ReactHashtag>
                    Hello #there fellow #dude#Works?
                </ReactHashtag>
            </div>
        );

        expect(context.html()).toMatchSnapshot();
    });

    it('Should render custom hashtags', () => {
        const hashtagRenderer = (hashtag) => (
            <div>
                {hashtag}
            </div>
        );

        const context = shallow(
            <div>
                <ReactHashtag renderHashtag={hashtagRenderer}>
                    #lots
                    #of
                    #hashtags
                    ÔºÉü§∑üèø‚Äç‚ôÄÔ∏è
                    #thatWasAUnicodeSet
                </ReactHashtag>
            </div>
        );

        expect(context.html()).toMatchSnapshot();
    });

    it('Should render hashtags with custom regex', () => {

        const context = shallow(
            <div>
                <ReactHashtag hashtagRegex={/(?:(?<=\s)|^)#(\w*[a-zA-z]\w*)+\b(?!#)/gm}>
                    Hello #there fellow #dude#Works?
                </ReactHashtag>
            </div>
        );

        expect(context.html()).toMatchSnapshot();
    });

    it('Should handle the hashtag actions', () => {
        const onHashtagClick = jest.fn();
        const context = mount(
            <div>
                <ReactHashtag onHashtagClick={onHashtagClick}>
                    #lots
                    #of
                    #hashtags
                </ReactHashtag>
            </div>
        );

        context.find('span').first().simulate('click');
        expect(onHashtagClick).toHaveBeenCalledWith("#lots", expect.any(Object));
    });


    it('Should handle the hashtag action and custom renderer', () => {
        const onHashtagClick = jest.fn();

        const hashtagRenderer = (hashtag, action) => (
            <p key={hashtag} onClick={() => action(hashtag)}>
                {hashtag}
            </p>
        );

        const context = mount(
            <div>
                <ReactHashtag renderHashtag={hashtagRenderer} onHashtagClick={onHashtagClick}>
                    #lots
                    #of
                    #hashtags
                </ReactHashtag>
            </div>
        );

        context.find('p').first().simulate('click');
        expect(onHashtagClick).toHaveBeenCalledWith("#lots");
    });

});