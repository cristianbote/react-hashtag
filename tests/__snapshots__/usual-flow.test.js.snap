// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReactHashtag should behave Should render custom hashtags 1`] = `"<div><div>#lots</div> <div>#of</div> <div>#hashtags</div> <div>＃🤷🏿‍♀️</div> <div>#thatWasAUnicodeSet</div></div>"`;

exports[`ReactHashtag should behave Should render hashtags 1`] = `"<div>Hello <span>#there</span> fellow <span>#dude#Works?</span></div>"`;

exports[`ReactHashtag should behave Should render hashtags with custom regex 1`] = `"<div>Hello <span>#there</span> fellow #dude#Works?</div>"`;
